Unable to build with builtin_cling=OFF for LLVM 9

When building with #6385 I get the following error when building with `builtin_cling=OFF`. I haven't investigated in detail so it could be something wrong on my part:


```log
[ 96%] Generating G__Core.cxx, ../lib/Core.pcm
cd /home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/build-dir/core && /home/conda/feedstock_root/build_artifacts/debug_1613296575343/_build_env/bin/cmake -E env LD_LIBRARY_PATH=/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/build-dir/lib: /home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/build-dir/core/rootcling_stage1/src/rootcling_stage1 -v2 -f G__Core.cxx -cxxmodule -s /home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/build-dir/lib/libCore.so -excludePath /home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/root-source -excludePath /home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/build-dir/ginclude -excludePath /home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/build-dir/externals -excludePath /home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/build-dir/builtins -writeEmptyRootPCM -m _Builtin_intrinsics -mByproduct _Builtin_intrinsics -mByproduct ROOT_Foundation_Stage1_NoRTTI -mByproduct ROOT_Foundation_C -mByproduct ROOT_Rtypes -D__STDC_NO_COMPLEX__ -D__COMPLEX_H__ -I/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/build-dir/include -I/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/build-dir/ginclude -I/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/root-source/core/base/inc -I/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/root-source/core/foundation/inc -I/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/root-source/core/cont/inc -I/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/root-source/core/gui/inc -I/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/root-source/core/meta/inc -I/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/root-source/core/clib/inc -I/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/root-source/core/rint/inc -I/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/root-source/core/zip/inc -I/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/root-source/core/thread/inc -I/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/root-source/core/textinput/inc -I/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/root-source/core/clingutils/inc -I/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/root-source/core/base/v7/inc -I/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/root-source/core/foundation/v7/inc -I/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/root-source/core/unix/inc -I/home/conda/feedstock_root/build_artifacts/debug_1613296575343/_h_env/include ROOT/TErrorDefaultHandler.hxx ROOT/TSequentialExecutor.hxx ROOT/StringConv.hxx Buttons.h Bytes.h Byteswap.h KeySymbols.h MessageTypes.h Riostream.h Rtypes.h TApplication.h TAtt3D.h TAttAxis.h TAttBBox2D.h TAttBBox.h TAttFill.h TAttLine.h TAttMarker.h TAttPad.h TAttText.h TBase64.h TBenchmark.h TBuffer3D.h TBuffer3DTypes.h TBuffer.h TColor.h TColorGradient.h TDatime.h TDirectory.h TEnv.h TException.h TExec.h TFileCollection.h TFileInfo.h TFolder.h TInetAddress.h TMacro.h TMathBase.h TMD5.h TMemberInspector.h TMessageHandler.h TNamed.h TNotifyLink.h TObject.h TObjString.h TParameter.h TPluginManager.h TPoint.h TPRegexp.h TProcessID.h TProcessUUID.h TQClass.h TQCommand.h TQConnection.h TQObject.h TRedirectOutputGuard.h TRefCnt.h TRef.h TRegexp.h TRemoteObject.h TROOT.h TRootIOCtor.h TStopwatch.h TStorage.h TString.h TStringLong.h TStyle.h TSysEvtHandler.h TSystemDirectory.h TSystemFile.h TSystem.h TTask.h TThreadSlots.h TTime.h TTimer.h TTimeStamp.h TUri.h TUrl.h TUUID.h TVersionCheck.h TVirtualAuth.h TVirtualFFT.h TVirtualGL.h TVirtualMonitoring.h TVirtualMutex.h TVirtualPadEditor.h TVirtualPad.h TVirtualPadPainter.h TVirtualPerfStats.h TVirtualPS.h TVirtualQConnection.h TVirtualRWMutex.h TVirtualTableInterface.h TVirtualViewer3D.h TVirtualX.h ROOT/RDirectoryEntry.hxx ROOT/RIndexIter.hxx strlcpy.h snprintf.h strtok.h ROOT/TSeq.hxx TArrayC.h TArrayD.h TArrayF.h TArray.h TArrayI.h TArrayL64.h TArrayL.h TArrayS.h TBits.h TBtree.h TClassTable.h TClonesArray.h TCollection.h TCollectionProxyInfo.h TExMap.h THashList.h THashTable.h TIterator.h TList.h TMap.h TObjArray.h TObjectTable.h TOrdCollection.h TRefArray.h TRefTable.h TSeqCollection.h TSortedList.h TVirtualCollectionProxy.h ESTLType.h RStringView.h TClassEdit.h TError.h ROOT/RIntegerSequence.hxx ROOT/RLogger.hxx ROOT/RMakeUnique.hxx ROOT/RNotFn.hxx ROOT/RSpan.hxx ROOT/RStringView.hxx ROOT/span.hxx ROOT/TypeTraits.hxx TUnixSystem.h root_std_complex.h GuiTypes.h TApplicationImp.h TBrowser.h TBrowserImp.h TCanvasImp.h TClassMenuItem.h TContextMenu.h TContextMenuImp.h TControlBarImp.h TGuiFactory.h TInspectorImp.h TObjectSpy.h TToggleGroup.h TToggle.h TBaseClass.h TClassGenerator.h TClass.h TClassRef.h TClassStreamer.h TDataMember.h TDataType.h TDictAttributeMap.h TDictionary.h TEnumConstant.h TEnum.h TFileMergeInfo.h TFunction.h TFunctionTemplate.h TGenericClassInfo.h TGlobal.h TInterpreter.h TInterpreterValue.h TIsAProxy.h TListOfDataMembers.h TListOfEnums.h TListOfEnumsWithLock.h TListOfFunctions.h TListOfFunctionTemplates.h TMemberStreamer.h TMethodArg.h TMethodCall.h TMethod.h TProtoClass.h TRealData.h TSchemaHelper.h TSchemaRule.h TSchemaRuleSet.h TStatusBitsChecker.h TStreamerElement.h TStreamer.h TVirtualIsAProxy.h TVirtualRefProxy.h TVirtualStreamerInfo.h TVirtualArray.h TVirtualObject.h Getline.h /home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/root-source/core/base/inc/LinkDef.h
fatal error: module map file '/home/conda/feedstock_root/build_artifacts/debug_1613296575343/_h_env/include/cling/module.modulemap' not found
fatal error: module map file '/home/conda/feedstock_root/build_artifacts/debug_1613296575343/_h_env/include/cling/module.modulemap' not found
fatal error: module map file '/home/conda/feedstock_root/build_artifacts/debug_1613296575343/_h_env/include/cling/module.modulemap' not found
<<< cling interactive line includer >>>: fatal error: could not build module '_Builtin_intrinsics'
Error: Module '_Builtin_intrinsics' failed to load.
fatal error: module map file '/home/conda/feedstock_root/build_artifacts/debug_1613296575343/_h_env/include/cling/module.modulemap' not found
fatal error: module map file '/home/conda/feedstock_root/build_artifacts/debug_1613296575343/_h_env/include/cling/module.modulemap' not found
fatal error: module map file '/home/conda/feedstock_root/build_artifacts/debug_1613296575343/_h_env/include/cling/module.modulemap' not found
fatal error: module map file '/home/conda/feedstock_root/build_artifacts/debug_1613296575343/_h_env/include/cling/module.modulemap' not found
Error: Error loading the default rootcling header files.
make[3]: *** [core/CMakeFiles/G__Core.dir/build.make:481: core/G__Core.cxx] Error 1
make[3]: Leaving directory '/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/build-dir'
make[2]: *** [CMakeFiles/Makefile2:6505: core/CMakeFiles/G__Core.dir/all] Error 2
make[2]: Leaving directory '/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/build-dir'
make[1]: *** [CMakeFiles/Makefile2:6484: core/CMakeFiles/Core.dir/rule] Error 2
make[1]: Leaving directory '/home/conda/feedstock_root/build_artifacts/debug_1613296575343/work/build-dir'
make: *** [Makefile:762: Core] Error 2
```

## How cling was built

Cling is being built from https://github.com/vgvassilev/cling/tree/upgrade_llvm90 with two patches added:

* https://github.com/chrisburr/cling-feedstock/blob/8359ffc4aafde1fcd09ffa2caeae93600449d439/recipe/use-findpackage.patch (see #7176)
* https://github.com/chrisburr/cling-feedstock/blob/8359ffc4aafde1fcd09ffa2caeae93600449d439/recipe/Always-install-ClingTargets.patch

## Reproducer

This should work on any machine with docker installed:

```bash
git clone https://github.com/chrisburr/root-feedstock.git -b llvm90-install-error
cd root-feedstock
export CPU_COUNT=$(nproc)
./build-locally.py --debug linux_64_numpy1.16python3.8.____cpython
# Run the line printed which will be similar to
cd xxxx && source xxxx
source /home/conda/recipe_root/build.sh
make Core VERBOSE=1 -j1
```

--- a/interpreter/cling/CMakeLists.txt
+++ b/interpreter/cling/CMakeLists.txt
@@ -485,6 +485,7 @@ if (NOT LLVM_INSTALL_TOOLCHAIN_ONLY)
     PATTERN "CMakeFiles" EXCLUDE
     PATTERN "*.inc"
     PATTERN "*.h"
+    PATTERN "*.modulemap"
     )
 endif()
 

